homeassistant/components/sensor/linky.py

```diff
@@ -0,0 +1,82 @@
+"""
+Support for Linky.
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/linky/
+"""
+import logging
+import json
+import datetime
+import voluptuous as vol
+
+from dateutil.relativedelta import relativedelta
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pylinky==0.1.3']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'linky'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    from pylinky.client import LinkyClient
+    client = LinkyClient(username, password)
+
+    try:
+        client._get_httpsession()
+        client._post_login_page()
+
+        today = datetime.date.today()
+
+        client._data["daily"] = client._get_data_per_day(
```

Comment:
Throws `client._post_login_page` an exception if the login fails? If yes, please move this to the update method and pass the `client` to the sensor entity.